internal/rootcoord/quota_center.go

```diff
@@ -616,20 +616,37 @@ func (q *QuotaCenter) forceDenyWriting(errorCode commonpb.ErrorCode, cluster boo
 }
 
 // forceDenyReading sets dql rates to 0 to reject all dql requests.
-func (q *QuotaCenter) forceDenyReading(errorCode commonpb.ErrorCode) {
-	var collectionIDs []int64
-	for dbID, collectionIDToPartIDs := range q.readableCollections {
-		for collectionID := range collectionIDToPartIDs {
-			collectionLimiter := q.rateLimiter.GetCollectionLimiters(dbID, collectionID)
-			updateLimiter(collectionLimiter, GetEarliestLimiter(), internalpb.RateScope_Collection, dql)
-			collectionLimiter.GetQuotaStates().Insert(milvuspb.QuotaState_DenyToRead, errorCode)
-			collectionIDs = append(collectionIDs, collectionID)
+func (q *QuotaCenter) forceDenyReading(errorCode commonpb.ErrorCode, cluster bool, dbIDs []int64) {
+	if cluster {
+		var collectionIDs []int64
+		for dbID, collectionIDToPartIDs := range q.readableCollections {
+			for collectionID := range collectionIDToPartIDs {
+				collectionLimiter := q.rateLimiter.GetCollectionLimiters(dbID, collectionID)
+				updateLimiter(collectionLimiter, GetEarliestLimiter(), internalpb.RateScope_Collection, dql)
+				collectionLimiter.GetQuotaStates().Insert(milvuspb.QuotaState_DenyToRead, errorCode)
+				collectionIDs = append(collectionIDs, collectionID)
+			}
 		}
+
+		log.RatedWarn(30, "QuotaCenter force to deny reading",
+			zap.Int64s("collectionIDs", collectionIDs),
+			zap.String("reason", errorCode.String()))
 	}
 
-	log.Warn("QuotaCenter force to deny reading",
-		zap.Int64s("collectionIDs", collectionIDs),
-		zap.String("reason", errorCode.String()))
+	if len(dbIDs) > 0 {
+		for _, dbID := range dbIDs {
+			dbLimiters := q.rateLimiter.GetDatabaseLimiters(dbID)
+			if dbLimiters == nil {
```

Comment:
This check is redundant with the one in `updateLimiter`.

Reply:
In updateLimiter, unable to log some limiter information if the limiter is nil.  Here detailed information can be logged for the missing DB limiter.